qutebrowser/browser/webview.py

```diff
@@ -145,13 +146,26 @@ def __init__(self, win_id, parent=None):
         self.loadProgress.connect(lambda p: setattr(self, 'progress', p))
         objreg.get('config').changed.connect(self.on_config_changed)
 
+    @pyqtSlot()
+    def on_initial_layout_completed(self):
+        """Add url to history now that we have displayed something."""
+        history = objreg.get('web-history')
+        if not self._orig_url.matches(self.cur_url,
+                                      QUrl.UrlFormattingOption(0)):
+            # If the url of the page is different than the url of the link
+            # originally clicked, save them both.
+            history.add_url(self._orig_url.toDisplayString())
+        history.add_url(self.cur_url.toDisplayString(), self.title())
```

Comment:
`toDisplayString` has the problem of e.g. decoding spaces in the URL which is undesirable here. This should probably be `toString(QUrl.FullyEncoded | QUrl.RemovePassword)` instead.